### ISSUE TITLE
[a11y-defect-2]: After clicking "go back to edit" focus should be sent to the `h3`

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#61243]
## Point of contact

@joshkimux

## Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

## Details

Upon clicking `go back to edit` in the confirm your address page, focus is sent to the `h2` when it should be sent to the `h3`

Timestamp is at `:32`

https://github.com/department-of-veterans-affairs/va.gov-team/assets/14154792/283e9848-a03b-401d-bc36-7e7f7583bea6

## Reproduction steps

1. Log in as user vetsgov+test236@id.me and [reach the contact information page on the NoD flow](https://staging.va.gov/decision-reviews/board-appeal/request-board-appeal-form-10182/introduction) using Safari on iOS
2. Click "edit" under mailing address
3. Click "save" without making any changes to the address
4. Click `go back to edit`
5. Confirm focus is sent to the `h2` and not the `h3`

## Proposed solution or next steps

Focus should be sent to the unique `h3` of the edit mailing address page

## Type of issue

- [X] Screenreader
- [ ] Keyboard
- [X] Focus
- [ ] Headings
- [ ] Color
- [ ] Zoom
- [ ] Semantics
- [ ] Axe-core
- [ ] Something else

**Comment by Mottie** [comment]
Oh wait, I was wrong. The code controlling this would be within the profile code.

**Comment by adamwhitlock1** [comment]
@joshkimux and @mtcA6  I deployed a fix to staging for this ticket, and it can be validated now.

You should be able to log in, go to the form, edit the mailing address and get to the address validation screen, and upon cancelling, the focus should be set to the h3.

**Comment by joshkimux** [comment]
https://github.com/department-of-veterans-affairs/va.gov-team/assets/14154792/78b71bb4-5530-4695-a37c-b9bcb1932731

Confirming this has been fixed. Thank you so much @adamwhitlock1 !!
